tests/countries/test_antigua_and_barbuda.py

```diff
@@ -0,0 +1,221 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.antigua_and_barbuda import AntiguaAndBarbuda, AG, ATG
+from tests.common import CommonCountryTests
+
+
+class TestAntiguaAndBarbuda(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1955, 2050)
+        super().setUpClass(AntiguaAndBarbuda, years=years, years_non_observed=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(AntiguaAndBarbuda, AG, ATG)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(AntiguaAndBarbuda(years=1954))
+
+    def test_special_holidays(self):
+        self.assertHolidayName("Public Holiday", "1993-08-03")
+        self.assertHolidayName(
+            "State Funeral of the late The Honourable Charlesworth T. Samuel", "2008-02-19"
+        )
+        self.assertHolidayName(
+            "State Funeral of the late The Honourable Sir George Herbert Walter", "2008-03-18"
+        )
+        self.assertHolidayName(
+            "Day after the General Election",
+            "2018-03-22",
+            "2023-01-19",
+        )
```

Comment:
```suggestion
        for dt, name in (
            ("1993-08-03", "Public Holiday"),
            ("2008-02-19", "State Funeral of the late The Honourable Charlesworth T. Samuel"),
            ("2008-03-18", "State Funeral of the late The Honourable Sir George Herbert Walter"),
            ("2018-03-22", "Day after the General Election"),
            ("2023-01-19", "Day after the General Election"),
        ):
            self.assertHolidayName(name, dt)
```
Usually, we check only dates for special holidays, but checking names as well is a good idea (and worth extending to other countries). So let's do it a little more optimized. :) Later we'll need to add a method for this to `TestCase`.

Reply:
Hmm - I'm not really sure about this, maybe:
```python
        name_day_after_general_election = "Day after the General Election"
```
```python
            ("2018-03-22", name_after_election),
            ("2023-01-19", name_after_election),
        ):
```

Reply:
Yes, it's better.

Reply:
fixed.